worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/validator/FieldValidator.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019-2023 Open Text.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.worker.ingestion.validator;
+
+import com.hpe.caf.worker.document.DocumentWorkerDocument;
+import com.hpe.caf.worker.document.DocumentWorkerFailure;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+public final class FieldValidator implements FieldValidatorInterface
+{
+    private final ArrayList<String> allowedFields;
+
+    public FieldValidator(final String validationFile) throws IOException
+    {
+        final ValidationFileAdapter adapter = new ValidationFileAdapter(validationFile);
+        allowedFields = adapter.getFieldKeys();
```

Comment:
I'm feeling quite worried about using a list of regular expressions here, because of the performance implications.  We're going to need to evaluate the cost of this approach.  But to give it the best chance possible you should compile the regular expressions ahead of time.

    private final List<Pattern> allowedFieldPatterns;
    ...
    allowedFieldPatterns = adapter.getFieldKeys().stream().map(Pattern::compile).collect(Collectors.toList());

That way the regular expression compilation will only be happening once rather than each time it is used.

Your `isValidField()` method would then become:

    private boolean isValidField(final String fieldKey)
    {
        return allowedFieldPatterns
            .stream()
            .anyMatch(allowedFieldPattern -> allowedFieldPattern.matcher(fieldKey).matches());
    }